src/components/StorageOverview/TopConsumers/TopConsumers.js

```diff
@@ -90,30 +98,129 @@ const TopConsumersBody = ({ topConsumerStats }) => {
   return <div>{results}</div>;
 };
 
-export const TopConsumers = ({ topConsumerStats, topConsumerLoaded, LoadingComponent }) => (
-  <DashboardCard>
-    <DashboardCardHeader>
-      <DashboardCardTitle>Top Projects by Requested Capacity</DashboardCardTitle>
-    </DashboardCardHeader>
-    <DashboardCardBody isLoading={!topConsumerLoaded} LoadingComponent={LoadingComponent}>
-      <TopConsumersBody topConsumerStats={topConsumerStats} />
-    </DashboardCardBody>
-  </DashboardCard>
-);
+export class TopConsumers extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: metricTypes[0],
+      sortBy: sortBy[0],
+    };
+  }
+
+  changeState = (key, newVal) => {
+    this.setState({ [key]: newVal }, this.getCurrentMetric);
+  };
+
+  getCapacity = data => get(data, 'data.result', []);
+
+  getCurrentMetric = () => {
+    switch (this.state.type) {
+      case PROJECTS:
+        switch (this.state.sortBy) {
+          case BY_USED_CAPACITY:
+            return this.getCapacity(this.props.projectsUsedCapacity);
+          case BY_REQUESTED_CAPACITY:
+            return this.getCapacity(this.props.projectsRequestedCapacity);
+          default:
+            // eslint-disable-next-line no-console
+            console.log(`Unknown metric ${this.state.sortBy.name}`);
+            return [];
+        }
+
+      case STORAGE_CLASSES:
+        switch (this.state.sortBy) {
+          case BY_USED_CAPACITY:
+            return [];
+          case BY_REQUESTED_CAPACITY:
+            return this.getCapacity(this.props.slClassesRequestedCapacity);
+          default:
+            // eslint-disable-next-line no-console
+            console.log(`Unknown metric ${this.state.sortBy.name}`);
+            return [];
+        }
+
+      case PODS:
+        switch (this.state.sortBy) {
+          case BY_USED_CAPACITY:
+            return this.getCapacity(this.props.podsUsedCapacity);
+          case BY_REQUESTED_CAPACITY:
+            return [];
+          default:
+            // eslint-disable-next-line no-console
+            console.log(`Unknown metric ${this.state.sortBy.name}`);
```

Comment:
not sure if these console statements are really necessary? Can another value really occur here? Isn't this in our control?

Reply:
Agreed. Can be removed.